packages/components/src/Switch/Switch.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { LegacyRef, RefObject, useRef } from 'react';
+import { useFocusRing } from '@react-aria/focus';
+import { useSwitch } from '@react-aria/switch';
+import { VisuallyHidden } from '@react-aria/visually-hidden';
+import { useToggleState } from '@react-stately/toggle';
+import { ToggleProps } from '@react-types/checkbox';
+import { AriaSwitchProps } from '@react-types/switch';
+
+import { ComponentProps } from '@marigold/types';
+import { conditional } from '@marigold/system';
+
+import { Box } from '../Box';
+import { Label } from '../Label';
+
+// Theme Extension
+// ---------------
+export interface SwitchThemeExtension<Value> {
+  switch?: {
+    [key: string]: Value;
+  };
+}
+
+// Props
+// ---------------
+export type SwitchProps = {
+  variant?: string;
+  labelVariant?: string;
+  disabled?: boolean;
+} & AriaSwitchProps &
+  ToggleProps &
+  ComponentProps<'input'>;
+
+// Component
+// ---------------
+export const Switch: React.FC<SwitchProps> = ({
+  variant = '',
+  labelVariant = 'above',
+  disabled,
+  ...props
+}) => {
+  const state = useToggleState(props);
+  const ref = useRef<HTMLInputElement>();
+  const { inputProps } = useSwitch(
+    props,
+    state,
+    ref as RefObject<HTMLInputElement>
+  );
+  const { focusProps } = useFocusRing();
+
+  return (
+    <Label variant={labelVariant}>
+      {props.children}
+      <VisuallyHidden>
+        <input
+          {...inputProps}
+          {...focusProps}
+          disabled={disabled}
+          ref={ref as LegacyRef<HTMLInputElement>}
+        />
+      </VisuallyHidden>
+      <Box
+        as="svg"
+        __baseCSS={{
```

Comment:
add `userSelect: 'none'` so text doesn't selected on accident

Reply:
<img width="274" alt="image" src="https://user-images.githubusercontent.com/985701/156555916-87dc08b5-bf45-4cb1-b4c0-c6418bb84c86.png">

Reply:
No in the baseCSS prop :)

Reply:
but this has no effect

Reply:
I've tried it like this but no effect: https://css-tricks.com/almanac/properties/u/user-select/